crates/ruff_linter/src/rules/pylint/rules/boolean_chained_comparison.rs

```diff
@@ -0,0 +1,145 @@
+    if expr_bool_op.op != BoolOp::And {
+        return;
+    }
+
+    // early exit when not all expressions are compare expressions
+    // TODO: check if this can even happen?
+    if !expr_bool_op.values.iter().all(Expr::is_compare_expr) {
+        return;
+    }
+
+    // retrieve all compare statements from expression
+    let compare_exprs: Vec<&ExprCompare> = expr_bool_op
+        .values
+        .iter()
+        .map(|stmt| stmt.as_compare_expr().unwrap())
+        .collect();
+
+    let results: Vec<Result<(), BooleanChainedComparison>> = compare_exprs
+        .iter()
+        .tuple_windows::<(&&ExprCompare, &&ExprCompare)>()
+        .filter(|(left_compare, right_compare)| {
+            are_compare_expr_simplifiable(left_compare, right_compare)
+        })
+        .map(|(left_compare, right_compare)| {
+            let Expr::Name(left_compare_right) = left_compare.comparators.first().unwrap() else {
+                return Ok(());
+            };
+
+            let Expr::Name(ref right_compare_left) = right_compare.left.as_ref() else {
+                return Ok(());
+            };
+
+            if left_compare_right.id() != right_compare_left.id() {
+                return Ok(());
+            }
+
+            Err(BooleanChainedComparison {
+                variable: left_compare_right.id().clone(),
+                range: TextRange::new(left_compare.range.start(), right_compare.range.end()),
+                replace_range: TextRange::new(
+                    left_compare_right.range.start(),
+                    right_compare_left.range.end(),
+                ),
+            })
+        })
+        .collect();
+
+    let results: Vec<BooleanChainedComparison> = results
+        .into_iter()
+        .filter(Result::is_err)
+        .map(|result| result.err().unwrap())
+        .collect();
+
+    checker.diagnostics.extend(
+        results
+            .into_iter()
+            .map(|result| {
+                let variable = result.variable.clone();
+                let range = result.range;
+                let replace_range = result.replace_range;
+                let mut diagnostic = Diagnostic::new(result, range);
+                diagnostic.set_fix(Fix::safe_edit(Edit::range_replacement(
+                    variable.to_string(),
+                    replace_range,
+                )));
+                diagnostic
+            })
+            .collect::<Vec<Diagnostic>>(),
+    );
+}
+
+/// Checks whether two compare expressions are simplifiable
+fn are_compare_expr_simplifiable(left: &ExprCompare, right: &ExprCompare) -> bool {
+    // only allow simplifying simple compare operations
+    if left.ops.len() != 1 || right.ops.len() != 1 {
+        return false;
+    }
+
+    matches!(
+        (left.ops.first().unwrap(), right.ops.first().unwrap()),
+        (CmpOp::Lt | CmpOp::LtE, CmpOp::Lt | CmpOp::LtE)
+            | (CmpOp::Gt | CmpOp::GtE, CmpOp::Gt | CmpOp::GtE)
+    )
```

Comment:
```suggestion
		let left_operator = left.ops.as_slice() else {
			return false;
		}
		
		let right_operator = right.ops.as_slice() else {
			return false;
		}

        matches!(
        (left_operator, right_operator),
        (CmpOp::Lt | CmpOp::LtE, CmpOp::Lt | CmpOp::LtE)
            | (CmpOp::Gt | CmpOp::GtE, CmpOp::Gt | CmpOp::GtE)
    )
```

Reply:
First this mentions this:

```
error[E0599]: no method named `as_slice` found for struct `std::boxed::Box<[ruff_python_ast::CmpOp]>` in the current scope
```

Then I import the trait: `use core::slice::SlicePattern;` for it to work, I get the following error:

```
error[E0658]: use of unstable library feature 'slice_pattern': stopgap trait for slice patterns
```

Reply:
Oh sorry, I incorrectly assumed that `left.ops` is a `Vec` but it's a `Box<[]>`. The following should work (dereference the Box to `[..]` and then take a reference of it):

```rust
let left_operator = &*left.ops else {
      return false;
  };
```

Reply:
I like it, implemented here, but adjusted slightly to match the single operation in the array:

```rs
let [left_operator] = &*left.ops else {
    return false;
};
```

Reply:
oh my sorry. Yeah my code is just wrong. It needs the `[left_operator]`.